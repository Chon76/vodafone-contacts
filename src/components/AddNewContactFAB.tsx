import { Fab, Icon } from 'native-base';
import { Entypo } from '@expo/vector-icons';

import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { useNavigation } from '@react-navigation/native';

import { RootStackParamList } from '../navigation/types';

export default function AddNewContactFAB() {
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  return (
    <Fab
      onPress={() => navigation.push('AddNewContact')}
      renderInPortal={false}
      shadow={2}
      size={16}
      icon={<Icon color="white" as={Entypo} name="plus" size="xl" />}
      bg={'primary'}
      testID={'fab'}
    />
  );
}
