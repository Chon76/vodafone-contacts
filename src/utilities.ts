import { Contact } from './types/Contact';

// eslint-disable-next-line import/prefer-default-export
export function getInitialLeters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  console.log('initials', initialLetters);

  return initialLetters;
}
