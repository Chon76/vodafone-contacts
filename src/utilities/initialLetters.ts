import { Contact } from '../types/Contact';

/**
 *
 * @param contacts an array of {@link Contact}
 * @returns the initial letters of the names of the contact
 */
// eslint-disable-next-line import/prefer-default-export
export function getInitialLetters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  return initialLetters;
}
