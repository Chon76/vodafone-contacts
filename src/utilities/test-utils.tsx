/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NativeBaseProvider } from 'native-base';
import { NavigationContainer } from '@react-navigation/native';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

const wrapper = ({ children }: { children: React.ReactNode }) => {
  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      <NavigationContainer>{children}</NavigationContainer>
    </NativeBaseProvider>
  );
};

// wrap tests with necessery providers
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper, ...options });

export * from '@react-navigation/native';
export { customRender as render };
