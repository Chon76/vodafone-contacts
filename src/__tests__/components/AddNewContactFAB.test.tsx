import { fireEvent } from '@testing-library/react-native';

import AddNewContactFAB from '../../components/AddNewContactFAB';

import { render } from '../../utilities/test-utils';

describe('add new contact fab', () => {
  it('renders correctly', () => {
    const fab = render(<AddNewContactFAB />);

    expect(fab.toJSON()).toMatchSnapshot();
  });

  // failing because of type error
  it.skip('navigates to add contact screen', () => {
    const fab = render(<AddNewContactFAB />);

    fireEvent.press(fab.getByTestId('fab'));
  });
});
