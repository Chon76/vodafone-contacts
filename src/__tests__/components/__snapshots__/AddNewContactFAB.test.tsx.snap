// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add new contact fab renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={{}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": {
          "100": "#cffafe",
          "200": "#a5f3fc",
          "300": "#67e8f9",
          "400": "#22d3ee",
          "50": "#ecfeff",
          "500": "#06b6d4",
          "600": "#0891b2",
          "700": "#0e7490",
          "800": "#155e75",
          "900": "#164e63",
        },
        "borderRadius": 9999,
        "bottom": 16,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "position": "absolute",
        "right": 16,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.22,
        "shadowRadius": 2.22,
        "width": 64,
        "zIndex": 20,
      }
    }
    testID="fab"
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    >
      <Text />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
